### PR TITLE
chore: switch to using jemalloc on all major binaries. (v0.4)

### PR DESCRIPTION
**Pull request by dpc**
This is said to improve rocksdb memory usage
caused by memory fragmentation.

In my tests, it saves half of peak and sustained memory being used.
It needs more longer runtime testing, but the initial results are so
good there's no reason to wait landing it, IMO.

"Reverse-backported" it to https://github.com/fedimint/fedimint/pull/6357

### CODE REVIEW

File: fedimint-cli/src/main.rs [hunk 1]
```diff
@@ -1,5 +1,12 @@
 use fedimint_cli::FedimintCli;
 use fedimint_core::fedimint_build_code_version_env;
+#[cfg(not(target_env = "msvc"))]
+use jemallocator::Jemalloc;
+
+#[cfg(not(target_env = "msvc"))]
+#[global_allocator]
+// rocksdb suffers from memory fragmentation when using standard allocator
+static GLOBAL: Jemalloc = Jemalloc;
```
Comment (maan2003) [inline fedimint-cli/src/main.rs:9]: do even care windows? (msvc)

Comment (RCasatta) [reply fedimint-cli/src/main.rs:9]: if yes, probably the dep has to be gated something like so:

```
[target.'cfg(not(target_env = "msvc"))'.dependencies]
jemallocator = "0.5"
```

Comment (dpc) [reply fedimint-cli/src/main.rs:9]: We probably don't, we don't have any CI to verify anything about it. I just copied the boilerplate from jemalloc's README and I don't think it hurts.

Comment (dpc) [reply fedimint-cli/src/main.rs:9]: Will do it in https://github.com/fedimint/fedimint/pull/6357

**Comment by elsirion** [comment]
I'd like to merge to master first, then we can trigger backports to 0.5 and 0.4 from there.

**Comment by dpc** [comment]
OK, closing this one as redundant.

**Comment by elsirion** [comment]
I'm happy to use this PR, just didn't want to end up in a situation where we get it on 0.4 and master and e.g. forget 0.5 :see_no_evil: Since the automatic backport failed for 0.4 let's use this PR to fix #6384.

**Review by elsirion — APPROVED** [review]

**Pull request closed without merging**